shoop/core/models/product_media.py

```diff
@@ -82,13 +82,11 @@ def effective_title(self):
 
     @property
     def url(self):
-        if not self.public:
-            raise ValueError("`get_effective_url()` may not be used on non-public media")
-
-        if self.file_id:
-            return self.file.url
-        else:
+        if self.external_url:
             return self.external_url
+        if self.file:
+            return self.file.url
+        return "/"
```

Comment:
We talked about this yesterday. The non-raising fallback is good, since the product details page wants to show the media with url.

@tulimaki Though, maybe the fallback should be "" rather than "/", because then it would be possible to check if there is a url by simply iffing for non-emptyness. Agree?